### ISSUE TITLE
Develop a new weight-initialization scheme for YOLO-One

### ISSUE THREAD

**Issue by ibrandiay** [IAtrax/YOLO-One#9]
### Motivation / Problem
The current YOLO-One implementation re-uses the original YOLOv8 weight-initialization routine without change.  
During experiments we observed:

* **Slow convergence** on small custom datasets (< 5 k images).  
* **Numerical instability** when the model is trained from scratch with resolution ≠ 640 × 640 or with depth-width scales n/s/m/l/x that differ from the pretrained weights.  
* **Poor transfer** on medical or satellite imagery where the channel statistics diverge significantly from ImageNet.

These symptoms point to an initialization strategy that is tuned for generic ImageNet pretraining rather than for the one-shot, few-shot or variable-resolution scenarios that YOLO-One targets.

### Implementation Sketch
```python
class YOLOOneInit:
    def __init__(self, mean=None, std=None):
        self.mean = torch.tensor(mean or [0., 0., 0.])
        self.std  = torch.tensor(std  or [1., 1., 1.])

    def __call__(self, module):
        if isinstance(module, (Conv, CSP, SPPF)):
            # scaled He init
            nn.init.kaiming_normal_(module.conv.weight, a=math.sqrt(5))
            if module.conv.bias is not None:
                fan_in, _ = nn.init._calculate_fan_in_and_fan_out(module.conv.weight)
                bound = 1 / math.sqrt(fan_in)
                nn.init.uniform_(module.conv.bias, -bound, bound)
        elif isinstance(module, Detect):
            # last-layer bias init for one-shot detection
            nn.init.constant_(module.cv3[-1].bias, -math.log(3 / 1))  # 1 classes
```

Hook the initializer into `YOLO.__init__` via a flag `--init yolo-one`.

### Additional Context
- Paper reference: “Fixup Initialization: Residual Learning Without Normalization” (Zhang et al., 2019).  
- Branch: `feat/yolo-one-init` (to be created).  
- Target release: 0.5.0.

---

**Labels**: enhancement, training, good first issue  
**Assignees**: @IAtrax/maintainers
